ios/browser/api/password/brave_password_api.mm

```diff
@@ -0,0 +1,472 @@
+      consumer_callback;
+
+  void OnGetPasswordStoreResults(
+      std::vector<std::unique_ptr<password_manager::PasswordForm>> results)
+      override;
+
+  base::WeakPtrFactory<PasswordStoreConsumerIOS> weak_ptr_factory_{this};
+};
+
+base::WeakPtr<PasswordStoreConsumerIOS> PasswordStoreConsumerIOS::GetWeakPtr() {
+  return weak_ptr_factory_.GetWeakPtr();
+}
+
+void PasswordStoreConsumerIOS::OnGetPasswordStoreResults(
+    std::vector<std::unique_ptr<password_manager::PasswordForm>> results) {
+  std::move(consumer_callback).Run(std::move(results));
+  delete this;
+}
+
+#pragma mark - BravePasswordAPI
+
+@interface BravePasswordAPI () {
+  scoped_refptr<password_manager::PasswordStoreInterface> password_store_;
+}
+@end
+
+@implementation BravePasswordAPI
+
+- (instancetype)initWithPasswordStore:
+    (scoped_refptr<password_manager::PasswordStoreInterface>)passwordStore {
+  if ((self = [super init])) {
+    DCHECK_CURRENTLY_ON(web::WebThread::UI);
+
+    password_store_ = passwordStore;
+  }
+  return self;
+}
+
+- (void)dealloc {
+  password_store_ = nil;
+}
+
+- (bool)isAbleToSavePasswords {
+  // Returns whether the initialization was successful.
+  return password_store_->IsAbleToSavePasswords();
+}
+
+- (id<PasswordStoreListener>)addObserver:(id<PasswordStoreObserver>)observer {
+  return [[PasswordStoreListenerImpl alloc] init:observer
+                                   passwordStore:password_store_];
+}
+
+- (void)removeObserver:(id<PasswordStoreListener>)observer {
+  [observer destroy];
+}
+
+- (void)addLogin:(IOSPasswordForm*)passwordForm {
+  password_store_->AddLogin([self createCredentialForm:passwordForm]);
+}
+
+- (password_manager::PasswordForm)createCredentialForm:
+    (IOSPasswordForm*)passwordForm {
+  // Store a PasswordForm representing a PasswordCredential.
+  password_manager::PasswordForm passwordCredentialForm;
+
+  if (passwordForm.usernameElement) {
+    passwordCredentialForm.username_element =
+        base::SysNSStringToUTF16(passwordForm.usernameElement);
+  }
+
+  if (passwordForm.usernameValue) {
+    passwordCredentialForm.username_value =
+        base::SysNSStringToUTF16(passwordForm.usernameValue);
+  }
+
+  if (passwordForm.passwordElement) {
+    passwordCredentialForm.password_element =
+        base::SysNSStringToUTF16(passwordForm.passwordElement);
+  }
+
+  if (passwordForm.passwordValue) {
+    passwordCredentialForm.password_value =
+        base::SysNSStringToUTF16(passwordForm.passwordValue);
+  }
+
+  passwordCredentialForm.url = net::GURLWithNSURL(passwordForm.url);
+
+  if ([passwordForm.signOnRealm length] != 0) {
+    passwordCredentialForm.signon_realm =
+        base::SysNSStringToUTF8(passwordForm.signOnRealm);
+  } else {
+    passwordCredentialForm.signon_realm = passwordCredentialForm.url.spec();
+  }
+
+  if (passwordForm.dateCreated) {
+    passwordCredentialForm.date_created =
+        base::Time::FromNSDate(passwordForm.dateCreated);
```

Comment:
same as above, make sure `passwordForm.dateCreated` is non-nil before passing into `FromNSDate`